### DISCUSSION TITLE
Budibase Public API: Import full application

### DISCUSSION THREAD

**Discussion by CheshireNine** [Budibase/budibase#9051] [Help] [answered]
**Question**
Is it possible to import full application ( exported JSON blob text file) to Budibase via some Public Budibase API?

I can't find any existing Public API for this purpose:
_POST /api/public/v1/applications_ budibase endpoint allows to create only BLANK application with given name and url.

**Why I need this**
Our developers work on their local Budibase environments. After feature is done we want to store the application and then import this app version on several sandbox/preprod Budibase environments on release date. So we are trying to setup CI/CD process to 
1. unpublish/delete existing version of the application from Budibase environment
2. import new version of application (stored in git as exported JSON blob text file) to the same environment and publish it.

Right now we can do it only manually via Budibase UI, but it is inconvenient process as we need to do it on multiple environments.
I will be thankful for any advice and workarounds.

**Comment by melohagan** [comment, marked as answer]
Hey @CheshireNine 

I am not aware of any workarounds at the moment, but I can see the value in being able to do this via the API. 
I've created a feature request from this discussion: https://github.com/Budibase/budibase/issues/9126
